### ISSUE TITLE
Colima with containerd runtime not exposing docker.sock

### ISSUE THREAD

**Issue by harel-e** [abiosoft/colima#958]
### Description

I started colima with containerd runtime. I cannot connect to it with docker since it didn't expose a docker socket
When I performed the same with docker runtime, it works fine.

### Version

colima version 0.6.7
git commit: ba1be00e9aec47f2c1ffdacfb7e428e465f0b58a

runtime: docker
arch: aarch64
client: v24.0.7
server: v24.0.7
limactl version 0.19.1
qemu-img version 8.2.0
Copyright (c) 2003-2023 Fabrice Bellard and the QEMU Project developers

### Operating System

- [ ] macOS Intel <= 13 (Ventura)
- [ ] macOS Intel >= 14 (Sonoma)
- [ ] Apple Silicon <= 13 (Ventura)
- [X] Apple Silicon >= 14 (Sonoma)
- [ ] Linux

### Output of `colima status`

INFO[0000] colima [profile=test] is running using macOS Virtualization.Framework
INFO[0000] arch: aarch64
INFO[0000] runtime: containerd
INFO[0000] mountType: virtiofs

### Reproduction Steps

1. colima start --vm-type vz --runtime containerd --cpu 2 test
2. colima status

### Expected behaviour

I expect colima to expose the docker.sock in the host OS and have it visible in colima status command, just like with docker runtime

### Additional context

Sample output when using docker runtime

colima start --vm-type vz --runtime docker --cpu 2

colima status 

INFO[0000] colima is running using macOS Virtualization.Framework
INFO[0000] arch: aarch64
INFO[0000] runtime: docker
INFO[0000] mountType: virtiofs
INFO[0000] socket: unix:///Users/testing/.colima/default/docker.sock

**Comment by tvh** [comment]
I am having the same issue. Until this is resolved, it is still possible to talk to docker via SSH:
```
docker context create colima-ssh --docker "host=ssh://colima"
```

**Comment by tharwan** [comment]
Same issue here with 0.6.9

**Comment by binary-signal** [comment]
Same issue here with 0.6.9, I suspect the issue occurs when using containerd + vz vm on mac os m1

**Comment by shucking** [comment]
Same issue: m3 macos, colima 0.6.10

**Comment by abiosoft** [comment]
This is actually the expected behaviour for containerd runtime. Do you still want to keep using docker for containerd runtime?

**Comment by tvh** [comment]
> This is actually the expected behaviour for containerd runtime. Do you still want to keep using docker for containerd runtime?

Yes. If I'm using the docker runtime _and_ some kubernetes, docker is full of the kubernetes containers, making it a bit hard to work with. And `vz` is necessary for `virtiofs`.

**Comment by elaijuh** [comment]
Maybe I am missing something here, but why do you want to use docker with containerd runtime? nerdctl is there

**Comment by Raboo** [comment]
> Maybe I am missing something here, but why do you want to use docker with containerd runtime? nerdctl is there

Not cool man! Not cool!
Why does people feel the need to go into questions like "Why do you need this? Why don't you do it my way?".
These type of questions are neither helpful or meaningful. Sure, offering an alternative is good, but everyone that installed colima knows about nerdctl. The README clearly refers to it. But the README also shows a video of using the docker cli with colima. So clearly there is a use case for it, using ssh instead of docker socket is a work-around, not a solution.

But to humor you.
There are literally dozens if not more of popular tools that use the docker socket and expect it, build tools, ci tools and a bunch of other dev tools. There are even more tools that rely on the docker cli.

**Comment by Raboo** [comment]
The aggression comes from the fact that @abiosoft already asked:
> Do you still want to keep using docker for containerd runtime?

After someone already answered yes to that, @elaijuh is is then questioning that answer and basically in nicer words stupidifying the person that answered yes and indirectly saying that he shouldn't use docker cli...
At least that is how it felt to me with the comment "nerdctl is there". It is a snide comment and there is where the aggression started.

I've seen this behavior before, and it discourages people. So after seeing that I just wanted to shoot it down directly.
But you are right in understanding the why, and ignoring the snide comment would have been the best course of action.

With that said, I won't continue this discussion any more in this issue to avoid hijacking it. If you feel the need to have a last word or discussing further in this topic, then I'm on Kubernetes slack. But really, I don't want to bother.

 I apologize for falling into the troll trap.

**Comment by binary-signal** [comment]
Lots of software, for better or worse, expects a docker.sock file to work with, even though we are using containerd. One good example of this is JetBrains IDEs.

**Comment by lloeki** [comment]
One reason would be tooling that squarely relies on the `docker` command.

Docker itself can make use of `containerd` as an image store, which brings the possibility to build multi-arch images (`docker buildx build --platform linux/x86_64,linux/aarch64 ...`), via this in `/etc/docker/daemon.json`:

```
    "features": {
      "containerd-snapshotter": true
    }
```

I'd argue that not only should a Docker context for colima be exposed by default, but that `containerd` should be the image store for Docker by default as well.

**Comment by utensil** [comment]
To add a use case for exposing `docker.sock` , I found [ctop](https://github.com/bcicen/ctop) not working for colima, and searching issues lead me here.

My `colima status` is

```
INFO[0000] colima is running using macOS Virtualization.Framework
INFO[0000] arch: aarch64
INFO[0000] runtime: docker
INFO[0000] mountType: virtiofs
INFO[0000] socket: unix:///Users/utensil/.colima/default/docker.sock
```

(i.e. I used `--vm-type=vz --vz-rosetta` to start colima, and I'm on Mac MIni M4, but I'm not using containerd runtime, so a little OT sorry, feel free to collapse my comment, but hopefully there might be useful information debugging the issue for containerd)

`ctop` complains: `Get "http://unix.sock/info": dial unix /var/run/docker.sock: connect: no such file or directory`.

 From `docker context ls` we can see:

```
NAME       DESCRIPTION                               DOCKER ENDPOINT                                     ERROR
colima *   colima                                    unix:///Users/utensil/.colima/default/docker.sock
default    Current DOCKER_HOST based configuration   unix:///var/run/docker.sock
```

that's why `docker` works fine for me (it found colima as the default context), but not `ctop` (which insists on the default docker.sock location).

EDIT: By RTFM, I found `ctop` works with colima via something like `DOCKER_HOST=unix:///Users/utensil/.colima/default/docker.sock ctop` or

```
DOCKER_HOST=`docker context ls --format json|grep colima|jq -r '.DockerEndpoint'` ctop
```

**Comment by abiosoft** [comment]
I am trying to understand the scenario better. Is it that you primarily need Docker but want to take advantage of Containerd features?
Or you want to interract with both nerdctl and docker cli?

**Comment by paulswartz** [comment]
Speaking for myself, I was trying to use the [attestations](https://docs.docker.com/build/metadata/attestations/) features of Docker locally, which requires the containerd image store.

**Comment by abiosoft** [comment]
@lloeki @paulswartz containerd image store is now enabled by default and the containerd socket is also forwarded https://github.com/abiosoft/colima/pull/1360.

You can try it out now by installing the development version with `brew install --head colima`.

Also, you would need to either delete existing instance and recreate, or run `colima update` to get a newer and supported docker version.

To @harel-e and everyone else, what is left to close this issue?
- containerd socket is now exposed, visible in the output of `colima status`.
- docker runtime is utilising the containerd image store (and by default).

As for the initial request of exposing docker.sock for containerd runtime, I think the inverse is the intention, to get containerd features with docker runtime like Docker is doing https://www.docker.com/blog/extending-docker-integration-with-containerd/.

**Comment by paulswartz** [comment]
`colima update` didn't work:

```
❯ colima update aarch64 -v
INFO[0000] refreshing package manager ...                context=docker
INFO[0008] checking for updates ...                      context=docker
WARN[0009] no updates available                          context=docker
```

But creating a new instance did!

**Comment by abiosoft** [comment]
A new release has been published https://github.com/abiosoft/colima/releases/tag/v0.8.3.
